x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/Expression.java

```diff
@@ -191,4 +193,16 @@ public String toString() {
     public String propertiesToString(boolean skipIfChild) {
         return super.propertiesToString(false);
     }
+
+    public boolean getVerifyOptimizedPlan() {
+        return verifyOptimizedPlan;
+    }
+
+    public void setVerifyOptimizedPlan() {
+        verifyOptimizedPlan = true;
+    }
+
+    public TypeResolution verifyOptimizedPlan() {
+        return TypeResolution.TYPE_RESOLVED;
+    }
```

Comment:
nit:
```suggestion
    /**
     * Some expressions need additional verifications that can only be performed after optimization.
    */
    public TypeResolution verifyAfterOptimization() {
        return TypeResolution.TYPE_RESOLVED;
    }
```

Reply:
The changes to this class are reverted in the last commit.